plugins/cilium-cni/main.go

```diff
@@ -590,12 +592,25 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 
 	// Specify that endpoint must be regenerated synchronously. See GH-4409.
 	ep.SyncBuildEndpoint = true
-	if err = c.EndpointCreate(ep); err != nil {
+	var epID *models.EndpointID
+	if epID, err = c.EndpointCreate(ep); err != nil {
 		logger.WithError(err).WithFields(logrus.Fields{
 			logfields.ContainerID: ep.ContainerID}).Warn("Unable to create endpoint")
 		return fmt.Errorf("unable to create endpoint: %s", err)
 	}
-
+	if epID == nil {
+		return fmt.Errorf("endpoint ID is nil")
+	}
+	if newEp, err := c.EndpointGet(epID.ID); err != nil {
+		logger.WithError(err).WithFields(logrus.Fields{
+			logfields.ContainerID: ep.ContainerID}).Warn("Unable to get endpoint")
+		return fmt.Errorf("unable to get endpoint: %s", err)
+	} else if newEp.Status.Networking.Mac != "" {
+		err = netns.ReplaceMacAddressWithLinkName(netNs, args.IfName, newEp.Status.Networking.Mac)
+		if err != nil {
+			return fmt.Errorf("unable to set MAC address on interface %s: %w", args.IfName, err)
+		}
+	}
```

Comment:
I think that this would be more compatible if the plugins do `if epID != nil { ... }` instead. This way, if the plugin version is mismatched with the agent version, the plugin will gracefully fall back to not implementing the MAC change.

Otherwise, I could see that if the user upgrades then downgrades Cilium, there could be a period where the agent becomes an older version, but the plugin is a new version, and every endpoint create would hit the `epID == nil` error case for some time.